crates/services/producer/src/block_producer.rs

```diff
@@ -382,6 +384,34 @@ where
     }
 }
 
+impl<ViewProvider, TxPool, Executor, GasPriceProvider, ConsensusProvider>
+    Producer<ViewProvider, TxPool, Executor, GasPriceProvider, ConsensusProvider>
+where
+    ViewProvider: HistoricalView + 'static,
+    ViewProvider::LatestView: BlockProducerDatabase,
+    Executor: ports::StorageReadReplayRecorder + 'static,
+    GasPriceProvider: GasPriceProviderConstraint,
+    ConsensusProvider: ConsensusParametersProvider,
+{
+    /// Re-executes an old block, getting the storage read events.
+    pub async fn storage_read_replay(
+        &self,
+        height: BlockHeight,
+    ) -> anyhow::Result<Vec<StorageReadReplayEvent>> {
+        let view = self.view_provider.latest_view()?;
+
+        let block = view.get_block(&height)?;
+        let transactions = block
+            .transactions()
+            .iter()
+            .map(|id| view.get_transaction(id).map(|tx| tx.into_owned()))
+            .collect::<Result<Vec<_>, _>>()?;
+        let block = block.into_owned().uncompress(transactions);
```

Comment:
I think you can a function `get_full_block` and it will be responsibility of the `fuel-core` to provide you a full `Block` type

Reply:
I moved it to the trait in 71a6a5607f, but it doesn't look much better. Did you have something else in mind?